esphome/components/display/display_buffer.cpp

```diff
@@ -643,14 +674,22 @@ Color Image::get_rgb565_pixel(int x, int y) const {
   auto r = (rgb565 & 0xF800) >> 11;
   auto g = (rgb565 & 0x07E0) >> 5;
   auto b = rgb565 & 0x001F;
-  return Color((r << 3) | (r >> 2), (g << 2) | (g >> 4), (b << 3) | (b >> 2));
+  Color color = Color((r << 3) | (r >> 2), (g << 2) | (g >> 4), (b << 3) | (b >> 2));
+  if (rgb565 == 1 && transparent_) {
```

Comment:
Ugh, this hurt to read. I think this would be much clearer if you specificed the (0,0,1) colour in hex or binary or something, anything other than decimal `1`. Ditto in the Python code.

```suggestion
  if (rgb565 == 0x0001 && transparent_) {
```